internal/pkg/machine/controllers/schematic.go

```diff
@@ -0,0 +1,53 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/cosi-project/runtime/pkg/state"
+
+	"github.com/siderolabs/talemu/internal/pkg/machine/machineconfig"
+	"github.com/siderolabs/talemu/internal/pkg/machine/runtime/resources/talos"
+	schematicsvc "github.com/siderolabs/talemu/internal/pkg/schematic"
+)
+
+func readCurrentSchematicID(ctx context.Context, r controller.Runtime) (string, error) {
```

Comment:
Moved this code from `ExtensionStatusController` to be also usable from `KernelCmdlineController`.